### ISSUE TITLE
Save game with Ctrl-S

### ISSUE THREAD

**Issue by gunchleoc** [singularity/singularity#285]
I'd like to be able to create savegames with `Ctrl-s` because I need to create a lot of them for making https://github.com/singularity/singularity/issues/283 and https://github.com/singularity/singularity/issues/284 reproducible (savegames need to be created *before* the bug happens).

Having to press the sequence `m s tab o y` every time is quite cumbersome, using the mouse takes even more time.

Is there a way to call `add_key_handler` in `map.py` while providing the key mod (`pygame.KMOD_CTRL`) and without having to change the key handling API?

**Comment by nthykier** [comment]
I think you can just register the K key and then explicitly checkf or the KMOD_CTRL modifier in the key event in the handler.

Alternatively, the sequence `m s enter enter` should do the same as `m s tab o y` but be significantly easier to type repeatedly.

**Comment by Xenega** [comment]
There a simple solution to handle KMOD_CTRL with add_key_handler. We could handle tuple. Basically we use (KMOD_CTRL, K_S) to create a shortcut to CTRL+K. First element would be the modifier, second the key. This allows to pass complex key to the event handler.

Alternately, we can add_mod_key_handler and hide the way the complex key are handled in the handler. This allows to change in the future the way they are stored.

Namedtuple and class are possible, but seems overkill to me.

**Comment by nthykier** [comment]
True, though the "slight" awkwardness in the code is secondary to me.

If we are to implement this feature, I am much more concerned about having a visual clue to the user that the game has been saved successfully (preferably without "stopping" the game).

**Comment by Xenega** [comment]
Yes, but we don't have a message system who is non-blocking. Something must be made. Preferentially simple and reserved for technical message. Basically, a label that disappeared after few second.

**Comment by nthykier** [comment]
For the record, I implemented a quick save feature via `F5`, which can then be (quick)loaded via `F9`.

Sadly there is no feedback on the quicksaves, which is annoying.

**Comment by nthykier** [comment]
Ok, sounds good. :)

Probably it will involve "on_tick" handling, which is currently tied to a dialog. Not sure how easy it will be to hook into from the notification component, but I also suspect this will be half the challenge.
